prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -475,6 +475,8 @@ private async Task CreateSelfDeclarationDocuments(int enrolleeId, ICollection<Se
                 return;
             }
 
+            var currentSelfDeclarationDocuments = await GetSelfDeclarationDocumentsAsync(enrolleeId) as IEnumerable<SelfDeclarationDocument>;
```

Comment:
I think you may have miss-understood my earlier comment, I wasn't advocating for making this method return Models instead of View Models. Since this method is being used to return data to the front end this should be returning view models.

Rather, why do a service call here when you can just do
```c#
var currentSelfDeclarationDocuments = await _context.SelfDeclarationDocuments
    .Where(document => document.EnrolleeId == enrolleeId)
    .ToListAsync();
```

_or_ in the `UpdateEnrolleeAsync` method, add the `.Include` for the self declaration documents then just pass them into this method.

Reply:
(also, why the cast to `IEnumerable<SelfDeclarationDocument>`? isn't it already returning that type?

Reply:
Yeah making the query in place is one of the 3 options I was looking at. The cast is again forgotten there, was testing something.

Reply:
The reason I did not want to go with the second option was to avoid code duplication since that query, in your comment, is identical to the query inside `GetSelfDeclarationDocumentsAsync` minus the `ProjectTo` part. That's why I thought we can do the projection outside the service method, for example if we're sending data to frontend, then do the projection right before sending the data to the frontend.